mlflow/store/db_migrations/versions/cfd24bdc0731_update_run_status_constraint_with_killed.py

```diff
@@ -0,0 +1,52 @@
+"""Update run status constraint with killed
+
+Revision ID: cfd24bdc0731
+Revises: 89d4b8295536
+Create Date: 2019-10-11 15:55:10.853449
+
+"""
+from alembic import op, context
+from mlflow.store.tracking.dbmodels.models import SqlRun
+from mlflow.entities import RunStatus
+
+# revision identifiers, used by Alembic.
+revision = 'cfd24bdc0731'
+down_revision = '2b4d017a5e9b'
+branch_labels = None
+depends_on = None
+
+
+new_run_status = [
+    RunStatus.to_string(RunStatus.SCHEDULED),
+    RunStatus.to_string(RunStatus.FAILED),
+    RunStatus.to_string(RunStatus.FINISHED),
+    RunStatus.to_string(RunStatus.RUNNING),
+    RunStatus.to_string(RunStatus.KILLED)
+]
+
+previous_run_status = [
+    RunStatus.to_string(RunStatus.SCHEDULED),
+    RunStatus.to_string(RunStatus.FAILED),
+    RunStatus.to_string(RunStatus.FINISHED),
+    RunStatus.to_string(RunStatus.RUNNING)
+]
+
+
+def _is_backend_sqlite():
+    return context.get_bind().engine.url.get_backend_name() == 'sqlite'
+
+
+def upgrade():
+    with op.batch_alter_table("runs") as batch_op:
+        if not _is_backend_sqlite():
```

Comment:
Sqlite3 seems to reflect CHECK constraints properly. Examining the `runs` table constraints for a sqlite database created against MLflow 1.4.0 yields the following result:
```
sqlite> select sql from sqlite_master where type='table' and name='runs';
CREATE TABLE runs (
        run_uuid VARCHAR(32) NOT NULL,
        name VARCHAR(250),
        source_type VARCHAR(20),
        source_name VARCHAR(500),
        entry_point_name VARCHAR(50),
        user_id VARCHAR(256),
        status VARCHAR(20),
        start_time BIGINT,
        end_time BIGINT,
        source_version VARCHAR(50),
        lifecycle_stage VARCHAR(20),
        artifact_uri VARCHAR(200),
        experiment_id INTEGER,
        CONSTRAINT run_pk PRIMARY KEY (run_uuid),
        CONSTRAINT source_type CHECK (source_type IN ('NOTEBOOK', 'JOB', 'LOCAL', 'UNKNOWN', 'PROJECT')),
        CONSTRAINT status CHECK (status IN ('SCHEDULED', 'FAILED', 'FINISHED', 'RUNNING')),
        CONSTRAINT runs_lifecycle_stage CHECK (lifecycle_stage IN ('active', 'deleted')),
        FOREIGN KEY(experiment_id) REFERENCES experiments (experiment_id)
)
```

Therefore, we probably shouldn't ignore sqlite when performing this upgrade. One caveat of upgrading the sqlite check constraint is that alembic will drop and recreate the entire table, and the new table will not reflect old check constraints unless they are explicitly specified via the `table_args` parameter of `op.batch_alter_table()`. More on this here: https://alembic.sqlalchemy.org/en/latest/batch.html#including-check-constraints

Reply:
Agree. Created a function to get all the constraints and pass it to table_args.

Instead of drop_create and create_constraint. I use now alter_column (as explained in the documentation in section just before including-check-constraints)